lambdas/src/core/resource/ratings/endpoint.rs

```diff
@@ -1,137 +1,126 @@
+pub fn routes() -> OpenApiRouter {
+    OpenApiRouter::new()
+        .routes(routes!(get_rating))
+        .routes(routes!(get_ratings))
+        .routes(routes!(post_rating))
+        .routes(routes!(get_ratings_city))
 }
 
+#[utoipa::path(
+  get,
+  path = "/ratings/{rating_id}",
+  description = "Get the details of a specific city rating",
+  tag = TAG,
+  params(
+    ("rating_id" = Uuid, Path, description = "Rating identifier")
+  ),
+  responses(
+    (status = OK, description = "Fetches the details of a city rating", body = Rating ),
+    ErrorResponses,
+  ))]
 async fn get_rating(
     Path(rating_id): Path<Uuid>,
-    component: Option<Query<BNAComponent>>,
     ctx: Context,
-) -> Result<Json<Value>, ExecutionError> {
-    let component = match component {
-        Some(c) => {
-            let Query(c) = c;
-            Some(c)
-        }
-        None => None,
-    };
-
-    get_rating_adaptor(rating_id, component, ctx)
+) -> Result<Json<Rating>, ExecutionError> {
+    get_rating_adaptor(rating_id, ctx)
         .await
         .map_err(|e| {
             debug!("{e}");
             e
         })
-        .map(|v| Json(json!(v)))
-}
-
-async fn get_ratings_city(
-    Path(rating_id): Path<Uuid>,
-    ctx: Context,
-) -> Result<Json<Value>, ExecutionError> {
-    get_ratings_city_adaptor(rating_id, ctx).await.map(Json)
+        .map(Rating::from)
+        .map(Json)
 }
 
-async fn get_ratings_analyses(
+#[utoipa::path(
+  get,
+  path = "/ratings",
+  description = "Get city ratings",
+  tag = TAG,
+  params(
+    ("page_size" = Option<u64>, Query, description = "The number of items per page", example = "25"),
+    ("page" = Option<u64>, Query, description = "The result page being returned", example = "1"),
+  ),
+  responses(
+    (status = OK, description = "Fetches the city ratings", body = Ratings),
+  ))]
+async fn get_ratings(
     pagination: Option<Query<PaginationParameters>>,
-) -> Result<Json<Value>, ExecutionError> {
+) -> Result<PageFlow<Ratings>, ExecutionError> {
     let Query(pagination) = pagination.unwrap_or_default();
-    get_ratings_analyses_adaptor(pagination.page, pagination.page_size())
-        .await
-        .map(|v| Json(json!(v.payload())))
-}
+    let (total_items, models) =
+        get_ratings_adaptor(pagination.page, pagination.page_size()).await?;
 
-async fn get_ratings_analysis(
-    Path(analysis_id): Path<Uuid>,
-    ctx: Context,
-) -> Result<Json<Value>, ExecutionError> {
-    get_ratings_analysis_adaptor(analysis_id, ctx)
-        .await
-        .map(Json)
+    let payload: Ratings = models.into();
+    Ok(PageFlow::new(
+        Paginatron::new(None, total_items, pagination.page, pagination.page_size()),
+        payload,
+    ))
 }
 
-async fn post_ratings_analysis(
-    Json(bna_pipeline): Json<BNAPipelinePost>,
-) -> Result<(StatusCode, Json<Value>), ExecutionError> {
-    post_ratings_analysis_adaptor(bna_pipeline)
+#[utoipa::path(
+  post,
```

Comment:
should this have a request body for the POST request?

Reply:
It totally should! I completely missed it 😱

Reply:
Fixed ✅